web3/web3/formatter.py

```diff
@@ -0,0 +1,192 @@
+import web3.utils.address as address
+import web3.utils.encoding as encoding
+import web3.utils.utils as utils
+import web3.utils.config as config
+from web3.eth.iban import Iban
+
+fromDecimal = encoding.fromDecimal
+toDecimal = encoding.toDecimal
+
+
+def isPredefinedBlockNumber(blockNumber):
+    return blockNumber in ["latest", "pending", "earliest"]
+
+
+def inputDefaultBlockNumberFormatter(blockNumber=None):
+    if not blockNumber:
+        return config.defaultBlock
+
+    return inputBlockNumberFormatter(blockNumber)
+
+
+def inputBlockNumberFormatter(blockNumber):
+    if not blockNumber:
+        return None
+    elif isPredefinedBlockNumber(blockNumber):
+        return blockNumber
+    return encoding.toHex(blockNumber)
+
+
+def inputCallFormatter(options):
+    """
+    Formats the input of a transaction and converts all values to HEX
+    """
+
+    options["from"] = options.get("from") or config["defaultAccount"]
+
+    if options.get("from"):
+        options["from"] = inputAddressFormatter(options["from"])
+
+    if options.get("to"):
+        options["to"] = inputAddressFormatter(options["to"])
+
+    for key in ["gasPrice", "gas", "value", "nonce"]:
+        if key in options:
+            options[key] = encoding.fromDecimal(options[key])
+
+    return options
+
+
+def inputTransactionFormatter(options):
+    """
+    Formats the input of a transaction and converts all values to HEX
+    """
+    options["from"] = options.get("from") or config["defaultAccount"]
+    options["from"] = inputAddressFormatter(options["from"])
+
+    if options.get("to"):
+        options["to"] = inputAddressFormatter(options["to"])
+
+    for key in ["gasPrice", "gas", "value", "nonce"]:
```

Comment:
*nitpick*

You could change this list to a `set` and make this a tiny bit faster (and slightly more correct in the purist sense)